overlord/state/change.go

```diff
@@ -139,16 +139,17 @@ const (
 // while the individual Task values would track the running of
 // the hooks themselves.
 type Change struct {
-	state              *State
-	id                 string
-	kind               string
-	summary            string
-	status             Status
-	clean              bool
-	data               customData
-	taskIDs            []string
-	ready              chan struct{}
-	lastObservedStatus Status
+	state                    *State
+	id                       string
+	kind                     string
+	summary                  string
+	status                   Status
+	clean                    bool
+	data                     customData
+	taskIDs                  []string
+	ready                    chan struct{}
+	lastObservedStatus       Status
+	lastRecordedNoticeStatus Status
```

Comment:
in principle to be fully consistent we would need to save this to disk otherwise we get potentially an extra notice if snapd restarts in between?

Reply:
Good point, Thanks! updated